### ISSUE TITLE
CAROs cyclicity prevents it being displayed correctly in OLS

### ISSUE THREAD

**Issue by matentzn** [obophenotype/caro#21]
See: https://github.com/EBISPOT/OLS/issues/324

If you want CARO to be displayed correctly, someone needs to go and figure out what has happened. This is often not trivial. Also, fixing this may also fixing some underlying issues with the modelling, so its probably worth it.

**Comment by matentzn** [comment]
Some help please @balhoff @dosumis 

Seems the problem is caused by "multi-modal cycles", i.e. cycles across different relations. 

```
BFO_0000001 (Entity)
	BFO_0000002 (continuant)
		BFO_0000004 (independent continuant)
			BFO_0000040 (material entity)
				CARO_0001010 (organism or virus or viroid)
					CARO_0010004 (Cellular organism)
			CARO_0030000 (biological entity) --> part of CARO_0010004 (Cellular organism)
				CARO_0000000 (anatomical entity)
					CARO_0000006 (Material anatomical Entity)
						CARO_0000003 (Connected anatomical structure)
							CARO_0010004 (Cellular organism)
```

As you can see 
- CARO_0010004 ---------------->CARO_0030000 and
- CARO_0030000 ----[part of]--->CARO_0010004

Can you comment on whether this is even conceptually sensible?

**Comment by balhoff** [comment]
I think it's totally valid logically. I'm surprised it's not a problem for OLS elsewhere; I would expect this kind of situation to show up in various ontologies.

But in this case I don't think the part_of relation is correct. It seems like there are biological entities that are not parts of cellular organisms (like virus structures).

**Comment by matentzn** [comment]
Yeah of course, these kinds of cycles are logically valid.. No I meant that particular part-of relation.

@udp is working on dealing with the OLS issue - that is just a bug after all.
